src/gui/sdl_mapper.cpp

```diff
@@ -3002,6 +3003,13 @@ void MAPPER_BindKeys(Section *sec)
 
 	if (SDL_GetModState()&KMOD_NUM)
 		MAPPER_TriggerEvent(num_lock_event, false);
+
+	if (!first_window) {
+		remove_window();
+		set_output(sec, wants_stretched_pixels());
+		GFX_ResetScreen();
+	}
+	first_window = false;
```

Comment:
Suggest moving this functionality into `sdlmain.cpp` packaged as a function, something like:

``` c++
void GFX_RegenerateWindow() {
	if (!first_window) {
		remove_window();
		set_output(sec, wants_stretched_pixels());
		GFX_ResetScreen();
	}
	first_window = false;
}
```

(Made available to the mapper with the prototype in `video.h`)

Reply:
Done.